### ISSUE TITLE
Support `ddev pull` for Platform.sh white-label partners

### ISSUE THREAD

**Issue by nicogommen** [ddev/ddev#6534]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem?

For users of a Platform.sh white-label partner, such as Ibexa Cloud, `ddev/ddev-platformsh` and `ddev pull platform` are currently not working. This is because `ddev pull platform` is using by default the `platform` CLI, whereas white-label partners have their own CLI. For instance for Ibexa Cloud, the CLI is `ibexa_cloud`, and can be installed via https://cli.ibexa.cloud/

### Describe your solution

The proposed solution would be to have a dedicated `ddev pull` command for each Platform.sh white-label partner using ddev.
For instance `ddev pull ibexa_cloud` for Ibexa Cloud.
It would do exactly the same as `ddev pull platform`, but using Ibexa Cloud CLI (`ibexa_cloud`) instead of `platform` CLI.

### Additional context

This test project can be used for Ibexa Cloud => https://console.ibexa.cloud/ibexa-cloud-internal-testing/4dz4gstt4twgo

**Comment by rfay** [comment]
You say that ddev/ddev-platformsh is not working... I thought it was working fine, the problem was `ddev pull platform` ?

**Comment by rfay** [comment]
IMO the best way to start an implementation here is to create an add-on (ddev-ibexa-cloud ?) that 
- [ ] Installs the required CLI using a .ddev/web-build/Dockerfile.ibexa-cloud
- [ ] Adds a .ddev/providers/ibexa-cloud.yaml that uses the  ibexa_cloud CLI.

**Comment by rfay** [comment]
Some one, I think you, said that it would be possible to make the `platform` cli behave exactly the same as ibexa_cloud. Could you reiterate that technique, and is it a reasonable and sustainable thing to do?

**Comment by nicogommen** [comment]
> You say that ddev/ddev-platformsh is not working... I thought it was working fine, the problem was `ddev pull platform` ?

Right, installing `ddev/ddev-platformsh` is working, it's `ddev pull platform` that is not (+ maybe any other command brought by `ddev/ddev-platformsh` that is using the CLI).

**Comment by rfay** [comment]
`ddev/ddev-platformsh` doesn't use the CLI at all. But if you have a concern about it, I'm all ears.

**Comment by pjcdawkins** [comment]
@rfay 
> Some one, I think you, said that it would be possible to make the platform cli behave exactly the same as ibexa_cloud. Could you reiterate that technique, and is it a reasonable and sustainable thing to do?

Reiterating:

```sh
# Get the Ibexa config YAML somehow, for example from an installed CLI:
ibexa_cloud self:config > /tmp/ibexa-cli-config.yaml

# Run the platform CLI substituting the config:
CLI_CONFIG_FILE=/tmp/ibexa-cli-config.yaml platform # ...
```

It's reasonable. Whether it's sustainable depends on that first step I suppose. The config is maintained in an internal GitLab file, and it can be exposed through the CLI's `self:config` file. Maybe we should host it online, like somewhere under https://cli.ibexa.co ?

The parts of the config that make the most (non-cosmetic difference) are the URLs and the `vendor_filter` under `api` - [example](https://github.com/platformsh/cli/blob/d7b6e8736e75e21ba1b9bf7bfeeb1e764e9d1724/internal/config/platformsh-cli.yaml#L52-L64).

**Comment by rfay** [comment]
* https://github.com/rfay/ddev-ibexa_cloud is starting to work (it can now pull database and files)
* I don't have a login to be able to use the `composer install` on the composer.json in the [test project](https://console.ibexa.cloud/ibexa-cloud-internal-testing/4dz4gstt4twgo)

**Comment by rfay** [comment]
https://github.com/ddev/ddev-ibexa-cloud is now available, `ddev get ddev/ddev-ibexa-cloud`. It has automated tests, and besides `ddev pull ibexa-cloud` it has `ddev ibexa_cloud`

@nicogommen if you can test and see if you like it we can move on to other white labels, or perhaps get some Ibexa folks using it first?

When the go CLI is appropriate for use here, we can change the Dockerfile install technique.

**Comment by rfay** [comment]
Haven't heard anything from you lately and don't know a path forward for Nestle, so this is not making progress.

**Comment by rfay** [comment]
I haven't been able to get a response on this issue for months (September, 2024), so it will be closed about March 1, 2025 if there's no interest from Platform.sh to continue with it.

**Comment by nicogommen** [comment]
Hey @rfay, I'm so sorry for the lack of answer on this one, I was caught up on many other things and then forgot about this ticket.
Let me try to test it today and get back to you.
Thanks for the work on this and the reminder.

**Comment by nicogommen** [comment]
@rfay , I did some tests and it works very well for me.
I will share it with Ibexa to make sure they edit their documentation (https://doc.ibexa.co/en/latest/ibexa_cloud/ddev_and_ibexa_cloud/).
Thanks again.

**Comment by rfay** [comment]
OK thanks @nicogommen - if you'll set up test targets for other white-label partners we'll implement the same thing for them. I do with there was a simpler setup, but oh well.

The users to give access to are `randy+platform (at) randyfay.com` and `accounts (at) ddev.com`

**Comment by rfay** [comment]
@nicogommen Someone changed the password for the accounts@ddev.com on ibexa-cloud, and turned on 2FA. Now tests fail, and we have no way to unpause the test project.

**Comment by nicogommen** [comment]
> [@nicogommen](https://github.com/nicogommen) Someone changed the password for the [accounts@ddev.com](mailto:accounts@ddev.com) on ibexa-cloud, and turned on 2FA. Now tests fail, and we have no way to unpause the test project.

That's strange. Let me see if I can find more information and help you with that.

Otherwise, we reached out to Ibexa about the new add-on, to make sure they update their documentation (https://doc.ibexa.co/en/latest/ibexa_cloud/ddev_and_ibexa_cloud/).
And they came back to us with some comments/questions. Should I share with them this ticket to bring them in this conversation? Or ask them to open a new one in the add-on repo (https://github.com/ddev/ddev-ibexa-cloud)?

Thanks.

**Comment by rfay** [comment]
There's already as ticket in the add-on:
* https://github.com/ddev/ddev-ibexa-cloud/issues/8

**Comment by nicogommen** [comment]
It's unrelated to the credentials and 2FA. It's questions/comments from Ibexa team about the add-on. I would like to have a place where where we can all discuss at the same place, rather than me transferring their questions to here and so on...
I will ask them to create a new ticket in https://github.com/ddev/ddev-ibexa-cloud.

**Comment by rfay** [comment]
Did this for ibexa-cloud, haven't heard of anything else, closing for now, happy to reopen if there's interest.
